src/main/java/ru/hogwarts/school/service/FacultyService.java

```diff
@@ -21,29 +24,35 @@ public Faculty addFaculty(Faculty faculty) {
     }
 
     public Faculty findFaculty(Long id) {
+        Optional<Faculty> faculty = facultyRepository.findById(id);
+        if (faculty.isPresent()) {
+            return faculty.get();
+        } else {
+            throw new NotFoundException();
+        }
 
-        return facultyRepository.findById(id).get();
     }
 
     public Faculty editFaculty(Faculty faculty) {
         return facultyRepository.save(faculty);
     }
 
-    public void deleteFaculty(Long id) {
+    public Faculty deleteFaculty(Long id) {
+        Faculty faculty = findFaculty(id);
         facultyRepository.deleteById(id);
+        return faculty;
     }
 
     public Collection<Faculty> getAllFaculties() {
         return facultyRepository.findAll();
     }
 
-    public Collection<Faculty> findByNameOrColor(String name, String color) {
-        return facultyRepository.findByNameIgnoreCaseOrColorIgnoreCase(name, color);
+    public Collection<Faculty> findByNameOrColor(String nameOrColor) {
+        return facultyRepository.findByNameIgnoreCaseOrColorIgnoreCase(nameOrColor, nameOrColor);
     }
 
-    public Faculty findByFacultyStudent(String name) {
-        return facultyRepository.findByStudentsNameIgnoreCase(name);
+    public Collection<Student> findByStudentsFaculty(Long id) {
+        return facultyRepository.findById(id).get().getStudents();
```

Comment:
здесь неправильно - если не будет факультета по id, то вызов get выкинет исключение - надо обработать
```suggestion
        return facultyRepository.findById(id)
                  .map(Faculty::getStudents)
                  .orElse(null);
```